chain-signatures/node/src/mesh/mod.rs

```diff
@@ -110,8 +121,31 @@ impl Mesh {
     }
 
     /// Ping the active participants such that we can see who is alive.
-    pub async fn ping(&mut self) {
+    async fn ping(&mut self) {
         self.active_participants = self.connections.ping().await;
         self.active_potential_participants = self.connections.ping_potential().await;
     }
+
+    pub async fn run(
+        mut self,
+        contract_state: Arc<RwLock<Option<ProtocolState>>>,
+        mesh_state: Arc<RwLock<MeshState>>,
+    ) -> anyhow::Result<()> {
+        let mut last_pinged = Instant::now();
+        loop {
+            if last_pinged.elapsed() > Duration::from_millis(300) {
+                if let Some(state) = contract_state.read().await.clone() {
+                    self.establish_participants(&state).await;
+                    let mut mesh_state = mesh_state.write().await;
+                    *mesh_state = MeshState {
```

Comment:
Why can't we fetch and set the updated info? Store all the information in one place and update it. Get rid of `establich_participants()`.